snuba/query/processors/mapping_optimizer.py

```diff
@@ -59,6 +67,12 @@ class MappingOptimizer(QueryProcessor):
     - `ifNull('tags.value[indexOf(tags.key, 'my_tag')]', '') = ''`
        this condition is equivalent to looking whether a tag is
        missing, which cannot be done with the hash map.
+        - There is a special case of this where the query will have
+            (
+                ifNull('tags.value[indexOf(tags.key, 'my_tag')]', '') != '' AND
```

Comment:
Just curious: why are we unable to apply this optimization if `ifNull` is absent?

Reply:
I've updated the comment. Let me know if there's any confusion left about how the optimization works.